Content.Client/Stealth/StealthSystem.cs

```diff
@@ -12,20 +14,46 @@ public sealed class StealthSystem : SharedStealthSystem
 {
     [Dependency] private readonly IPrototypeManager _protoMan = default!;
     [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IClientAdminManager _adminManager = default!;
+    [Dependency] private readonly ContainerSystem _containerSystem = default!;
 
     private ShaderInstance _shader = default!;
+    private ShaderInstance _altShader = default!;
+
+    private float timer = 0;
 
     public override void Initialize()
     {
         base.Initialize();
 
         _shader = _protoMan.Index<ShaderPrototype>("Stealth").InstanceUnique();
+        _altShader = _protoMan.Index<ShaderPrototype>("AccessibleFullStealth").InstanceUnique();
 
         SubscribeLocalEvent<StealthComponent, ComponentShutdown>(OnShutdown);
         SubscribeLocalEvent<StealthComponent, ComponentStartup>(OnStartup);
         SubscribeLocalEvent<StealthComponent, BeforePostShaderRenderEvent>(OnShaderRender);
     }
 
+    //don't like doing this but it's the only good way to have the shader update as far as I can tell
+    //though there's probably a better way, been staring at this for a while and kinda just want to be done with it
+    //it works well, but it's not a very elegant solution
```

Comment:
this is really scary but we can fix it at some point down the line (we are never going to fix this)

Reply:
it's not _actually_ that bad since it's doing basically nothing, it just felt wierd to do
I'm like 90% sure I could just move it into handling the component state now that I think about it